### PR TITLE
`Team exercises`: Fix missing open editor button when individual deadline set

### PR DESCRIPTION
**Pull request by EneaGore**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [X] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [X] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [X] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

Fixes issue [8610](https://github.com/ls1intum/Artemis/issues/8610)

### Description
When booleans are set whether a button should be shown or not individual deadline is not considered. Now it is included in all exercises.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Exercise of each type

DONT test on TS2 or TS3
1. Log in to Artemis
2. Create Team and individual Exercises
3. 1 Student/Team participates before deadline, 1 after deadline
4. Change individual deadline
5. Confirm that button to submit is available and submit button also looks and works correctly.

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

Before:
buttonMissingindividual
buttonMissingTeam
After:
buttonthere

### CODE REVIEW

**Review by SimonEntholzer — APPROVED** [review]
Code fix makes sense :+1:

File: src/main/webapp/app/exercises/modeling/participate/modeling-submission.component.ts [hunk 1]
```diff
@@ -216,7 +216,9 @@ export class ModelingSubmissionComponent implements OnInit, OnDestroy, Component
             this.modelingExercise &&
             !!this.modelingExercise.dueDate &&
             !!this.participation.initializationDate &&
-            dayjs(this.participation.initializationDate).isAfter(this.modelingExercise.dueDate);
+            dayjs(this.participation.initializationDate).isAfter(this.modelingExercise.dueDate) &&
+            dayjs(this.participation.initializationDate).isAfter(this.participation?.individualDueDate);
```
Comment (JohannesStoehr) [inline src/main/webapp/app/exercises/modeling/participate/modeling-submission.component.ts:220]:
```suggestion
            dayjs(this.participation.initializationDate).isAfter(getExerciseDueDate(this.modelingExercise, this.participation));
```

Comment (JohannesStoehr) [reply src/main/webapp/app/exercises/modeling/participate/modeling-submission.component.ts:220]: Please also adapt the other occurrences

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]

**Review by JohannesWt — DISMISSED** [review]
Tested in testing session - works as expected 👍

**Review by JohannesStoehr — APPROVED** [review]
Tested in Testing session. Worked as expected

**Review by az108 — DISMISSED** [review]
Works as expected on TS4. LGTM

**Review by coolchock — DISMISSED** [review]
tested in testing session, works as described

**Review by BBesrour — DISMISSED** [review]
Demonstrated in a testing session, also code lgtm

**Review by krusche — DISMISSED** [review]

**Pull request merged**